delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/ElasticSearchQuery.scala

```diff
@@ -0,0 +1,113 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.compositeviews
+
+import akka.http.scaladsl.model.Uri
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.CompositeViewProjection.ElasticSearchProjection
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.CompositeViewRejection.{AuthorizationFailed, ProjectionNotFound, WrappedElasticSearchClientError}
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.ProjectionType.ElasticSearchProjectionType
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.{CompositeView, CompositeViewRejection}
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.client.ElasticSearchClient
+import ch.epfl.bluebrain.nexus.delta.sdk.Acls
+import ch.epfl.bluebrain.nexus.delta.sdk.model.IdSegment
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SortList
+import ch.epfl.bluebrain.nexus.delta.sourcing.config.ExternalIndexingConfig
+import io.circe.{Json, JsonObject}
+import monix.bio.IO
+
+trait ElasticSearchQuery {
+
+  /**
+    * Queries the Elasticsearch index of the passed composite views' projection.
+    * We check for the caller to have the necessary query permissions on the views' projections before performing the query.
+    *
+    * @param id           the id of the composite view either in Iri or aliased form
+    * @param projectionId the id of the composite views' target projection either in Iri or aliased form
+    * @param project      the project where the view exists
+    * @param query        the elasticsearch query to run
+    * @param qp           the extra query parameters for the elasticsearch index
+    * @param sort         the sorting configuration
+    */
+  def query(
+      id: IdSegment,
+      projectionId: IdSegment,
+      project: ProjectRef,
+      query: JsonObject,
+      qp: Uri.Query,
+      sort: SortList
+  )(implicit caller: Caller): IO[CompositeViewRejection, Json]
+
+  /**
+    * Queries all the Elasticsearch indices of the passed composite views' projection.
+    * We check for the caller to have the necessary query permissions on the views' projections before performing the query.
+    *
+    * @param id      the id of the composite view either in Iri or aliased form
+    * @param project the project where the view exists
+    * @param query   the elasticsearch query to run
+    * @param qp      the extra query parameters for the elasticsearch index
+    * @param sort    the sorting configuration
+    */
+  def queryProjections(
+      id: IdSegment,
+      project: ProjectRef,
+      query: JsonObject,
+      qp: Uri.Query,
+      sort: SortList
+  )(implicit caller: Caller): IO[CompositeViewRejection, Json]
+
+}
+
+object ElasticSearchQuery {
+  final def apply(
+      acls: Acls,
+      views: CompositeViews,
+      client: ElasticSearchClient
```

Comment:
Same thing as its BG counterpart, if we can get rid of the client here that we will never test here anyway, testing is gonna be easier

Reply:
I'm keeping this for consistency with the other tests.
There could be a refactoring of all those tests where a function is enough, but I'm not going to do it in the current PR